torch/lib/THNN/generic/LogSoftMax.c

```diff
@@ -28,24 +36,26 @@ void THNN_(LogSoftMax_updateOutput)(
   uint64_t dim_stride = inner_size;
   uint64_t outer_stride = dim_size * dim_stride;
 
+  LOG_SOFTMAX_DEFINITIONS
+
 #pragma omp parallel for
-  for (uint64_t i = 0; i < outer_size * inner_size; i++)
+  LOG_SOFTMAX_FOR(i, 0, outer_size * inner_size)
```

Comment:
What's the problem again? I really dislike the current syntax plus, `i` and `d` will be shared across all threads now!!

Reply:
MSVC only supports OpenMP 2.4, which does not support index variable definitions and unsigned index variable in the for statement. I know this one is not good, but can't figure out a better one.

Reply:
It used to be done like that in [FeatureLPPooling.c](https://github.com/pytorch/pytorch/blob/master/torch/lib/THNN/generic/FeatureLPPooling.c) like this:
```Cpp
#ifdef _MSC_VER
  #define size_type int64_t
  #define cast_type (int64_t)
#else
  #define size_type size_t
  #define cast_type
#endif
// define index variables
size_t i; 
#pragma omp parallel for
for (i = 0; i < n; i++) {
// do sth.
}
```
But changing like this will cause the test of nn to fail.

Reply:
It's clear that it will cause it to fail, because `i` is shared between all OMP threads. You need to use `omp parallel for private(i)` to make it thread local